downloader/src/test/kotlin/DownloaderTest.kt

```
@@ -0,0 +1,145 @@
+                    version = "",
+                    declaredLicenses = sortedSetOf(),
+                    description = "",
+                    homepageUrl = "",
+                    downloadUrl = "",
+                    hash = "",
+                    hashAlgorithm = "",
+                    vcsPath = submoduleName,
+                    vcsProvider = "Git",
+                    vcsUrl = "git@github.com:heremaps/oss-review-toolkit.git",
+                    vcsRevision = ""
+            )
+            Main.download(pkg, outputDir)
+            val downloadedProjectDir = File(outputDir, pkg.normalizedName)
+            val downloadedModuleDir = File(downloadedProjectDir, submoduleName)
+
+            outputDir.list().size shouldBe 1
+            outputDir.list().first() shouldBe pkg.normalizedName
+
+            downloadedProjectDir.isDirectory shouldBe true
+            downloadedModuleDir.isDirectory shouldBe true
+
+            downloadedModuleDir.list().size shouldBe beGreaterThan(1)
+
+            val moduleDirList = downloadedModuleDir.list().asList()
+            moduleDirList should contain("build.gradle")
+            moduleDirList should contain("src")
+        }.config(tags = setOf(IntegrationTest))
+
+        "Downloads github oss-review-toolkit repo " {
+            val pkg = Package(
+                    packageManager = "Gradle",
+                    namespace = "",
+                    name = "oss-review-toolkit",
+                    version = "",
+                    declaredLicenses = sortedSetOf(),
+                    description = "",
+                    homepageUrl = "",
+                    downloadUrl = "",
+                    hash = "",
+                    hashAlgorithm = "",
+                    vcsPath = "",
+                    vcsProvider = "Git",
+                    vcsUrl = "git@github.com:heremaps/oss-review-toolkit.git",
+                    vcsRevision = ""
+            )
+            try {
```

Please do not swallow exceptions in tests.